Source/WTF/wtf/HashTable.h

```diff
@@ -635,51 +635,42 @@ DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER(HashTable);
     {
     }
 
-#if !ASSERT_ENABLED
-
-    template<typename Key, typename Value, typename Extractor, typename HashFunctions, typename Traits, typename KeyTraits>
-    template<typename HashTranslator, typename T>
-    inline void HashTable<Key, Value, Extractor, HashFunctions, Traits, KeyTraits>::checkKey(const T&)
-    {
-    }
-
-#else // ASSERT_ENABLED
-
-    template<typename Key, typename Value, typename Extractor, typename HashFunctions, typename Traits, typename KeyTraits>
+    template<typename Key, typename Value, typename Extractor, typename HashFunctions, typename Traits, typename KeyTraits, ShouldValidateKey validateKey>
     template<typename HashTranslator, typename T>
-    void HashTable<Key, Value, Extractor, HashFunctions, Traits, KeyTraits>::checkKey(const T& key)
+    inline void HashTable<Key, Value, Extractor, HashFunctions, Traits, KeyTraits, validateKey>::checkKey(const T& key)
     {
         if constexpr (std::is_convertible_v<T, Key>) {
-            ASSERT(!isHashTraitsEmptyValue<KeyTraits>(key));
-            ASSERT(!KeyTraits::isDeletedValue(key));
+            RELEASE_ASSERT(!isHashTraitsEmptyValue<KeyTraits>(key));
+            RELEASE_ASSERT(!KeyTraits::isDeletedValue(key));
         } else if constexpr (HashFunctions::safeToCompareToEmptyOrDeleted) {
-            ASSERT(!HashTranslator::equal(KeyTraits::emptyValue(), key));
+            RELEASE_ASSERT(!HashTranslator::equal(KeyTraits::emptyValue(), key));
 
             ALLOW_DEPRECATED_DECLARATIONS_BEGIN
             typename std::aligned_storage<sizeof(ValueType), std::alignment_of<ValueType>::value>::type deletedValueBuffer;
             ALLOW_DEPRECATED_DECLARATIONS_END
             ValueType* deletedValuePtr = reinterpret_cast_ptr<ValueType*>(&deletedValueBuffer);
             ValueType& deletedValue = *deletedValuePtr;
             Traits::constructDeletedValue(deletedValue);
-            ASSERT(!HashTranslator::equal(Extractor::extract(deletedValue), key));
+            RELEASE_ASSERT(!HashTranslator::equal(Extractor::extract(deletedValue), key));
         }
```

Comment:
If we're going to call this the safe HashMap, you need an else clause here:

```
} else {
    STATIC_ASSERT_NOT_REACHED_FOR_TYPE(Key, "ShouldValidateKey::Yes was specified, but no way to validate key.");
}
```

Eventually we should fill in how to validate keys where `!std::is_convertible_v<T, Key>)` and `! HashFunctions::safeToCompareToEmptyOrDeleted`. But a compile-time assertion is good enough to get started for now.